src/main/java/net/apnic/whowas/domain/controller/DomainSearchRouteController.java

```diff
@@ -0,0 +1,86 @@
+package net.apnic.whowas.domain.controller;
+
+import javax.servlet.http.HttpServletRequest;
+
+import net.apnic.whowas.error.MalformedRequestException;
+import net.apnic.whowas.history.ObjectClass;
+import net.apnic.whowas.history.ObjectIndex;
+import net.apnic.whowas.history.ObjectSearchIndex;
+import net.apnic.whowas.history.ObjectSearchKey;
+import net.apnic.whowas.rdap.controller.RDAPControllerUtil;
+import net.apnic.whowas.rdap.controller.RDAPResponseMaker;
+import net.apnic.whowas.rdap.TopLevelObject;
+import net.apnic.whowas.search.SearchResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Rest Controller for the RDAP /domains search path segment.
+ *
+ * This controller is not implemented and returns a HttpStatus code indicating
+ * such.
+ */
+@RestController
+@RequestMapping("/domains")
+public class DomainSearchRouteController
+{
+    private final static Logger LOGGER = LoggerFactory.getLogger(DomainSearchRouteController.class);
+
+    private final ObjectIndex objectIndex;
+    private final RDAPControllerUtil rdapControllerUtil;
+    private final ObjectSearchIndex searchIndex;
+
+    @Autowired
+    public DomainSearchRouteController(ObjectIndex objectIndex,
+        ObjectSearchIndex searchIndex, RDAPResponseMaker rdapResponseMaker)
+    {
+        this.objectIndex = objectIndex;
+        this.rdapControllerUtil = new RDAPControllerUtil(rdapResponseMaker);
```

Comment:
It was a conscious decision to do it this way for testing purposes. Originally it was setup like this but between Jim and myself we have changed it to what you see now.

Reply:
What problem is it solving using explicit creation of RDAPControllerUtil objects, one for each *Controller class?

Reply:
Getting rid of dependencies as I understand it.

Reply:
{comment} If RDAPControllerUtil had some state that you didn't want to get mixed up with other instances I could understand that, but there is still the same number of dependencies. Now RDAPControllerUtil has multiple instances and the controllers are now hardcoding their dependency on it.

Reply:
I think it makes sense here for the hard / static dependency here and to consider controller + RDAPControllerUtil as a unit, not everything has to be a bean. Suppose this class we implemented as a base class, we wouldn't be worried about removing the static dependency, nor would we be interested in the interactions between the base class and the child class - i see this particular composition as the same thing.

When testing the controllers, the interaction we want to test is with the service / index.

Reply:
As long as you are aware that you are choosing the tight coupling option here, and it's pros and cons. Despite it's name, I see RDAPControllerUtil as the "response formatter" for the controller because that's all it's doing. If it was more resource intensive (like, a lot more), or there was a need to think about other response formats then I would be leaving the suggestion tag in place (or removing all tags), but I changed it to comment because I can see that you are treating them as a single unit and just wanted to bring these ideas out for discussion. (Sorry about how, my communication skills suck)

Reply:
I think your point raises a good issue in that the name is probably not doing it justice anymore (left over from when the class had a bit more going on). Something like RDAPResponseFactory may be a better alternative now.